app/service/data_svc.py

```diff
@@ -151,14 +151,24 @@ async def load_ability_file(self, filename, access):
                 name = ab.pop('name', '')
                 description = ab.pop('description', '')
                 tactic = ab.pop('tactic', None)
-                technique_id = ab.get('technique', dict()).get('attack_id')
-                technique_name = ab.pop('technique', dict()).get('name')
+                if 'technique_id' in ab:
+                    technique_id = ab.pop('technique_id')
+                else:
+                    technique_id = ab.get('technique', dict()).get('attack_id')
+                if 'technique_name' in ab:
+                    technique_name = ab.pop('technique_name')
+                else:
+                    technique_name = ab.pop('technique', dict()).get('name')
                 privilege = ab.pop('privilege', None)
```

Comment:
each of these if/else to normalize the data should be moved to separate function/s if it's all to convert a prior version to new, then let's create a v0_conversion function so it's separate or convert the offenders within the plugins to conform

Reply:
Moved each one of these if/else blocks out to a separate conversion function, hopefully this is a bit cleaner.

Also fixed a bug that occurred when updating v0 ability files with PATCH - seems that both v0/v1 fields were being added to ability files, causing errors when the ability object was loaded